OrbitCore/SamplingProfiler.cpp

```diff
@@ -183,44 +182,43 @@ void SamplingProfiler::ProcessSamples(
 
   SortByThreadUsage();
 
-  FillThreadSampleDataSampleReports();
+  FillThreadSampleDataSampleReports(capture_data);
 }
 
-void SamplingProfiler::ResolveCallstacks(
-    const CallstackData* callstack_data,
-    absl::flat_hash_map<uint64_t, LinuxAddressInfo>* address_infos) {
-  callstack_data->ForEachUniqueCallstack([this, address_infos](const CallStack& call_stack) {
-    // A "resolved callstack" is a callstack where every address is replaced
-    // by the start address of the function (if known).
-    std::vector<uint64_t> resolved_callstack_data;
-
-    for (uint64_t address : call_stack.GetFrames()) {
-      if (exact_address_to_function_address_.find(address) ==
-          exact_address_to_function_address_.end()) {
-        UpdateAddressInfo(address, address_infos);
-      }
-
-      auto address_it = exact_address_to_function_address_.find(address);
-      if (address_it != exact_address_to_function_address_.end()) {
-        const uint64_t& function_address = address_it->second;
-        resolved_callstack_data.push_back(function_address);
-        function_address_to_callstack_[function_address].insert(call_stack.GetHash());
-      } else {
-        resolved_callstack_data.push_back(address);
-      }
-    }
-
-    CallStack resolved_callstack(std::move(resolved_callstack_data));
-
-    CallstackID resolved_callstack_id = resolved_callstack.GetHash();
-    if (unique_resolved_callstacks_.find(resolved_callstack_id) ==
-        unique_resolved_callstacks_.end()) {
-      unique_resolved_callstacks_[resolved_callstack_id] =
-          std::make_shared<CallStack>(resolved_callstack);
-    }
-
-    original_to_resolved_callstack_[call_stack.GetHash()] = resolved_callstack_id;
-  });
+void SamplingProfiler::ResolveCallstacks(const CaptureData& capture_data) {
+  capture_data.GetCallstackData()->ForEachUniqueCallstack(
+      [this, &capture_data](const CallStack& call_stack) {
+        // A "resolved callstack" is a callstack where every address is replaced
+        // by the start address of the function (if known).
+        std::vector<uint64_t> resolved_callstack_data;
+
+        for (uint64_t address : call_stack.GetFrames()) {
+          if (exact_address_to_function_address_.find(address) ==
+              exact_address_to_function_address_.end()) {
+            MapAddressToFunctionAddress(address, capture_data);
+          }
+
+          auto address_it = exact_address_to_function_address_.find(address);
+          if (address_it != exact_address_to_function_address_.end()) {
+            const uint64_t& function_address = address_it->second;
+            resolved_callstack_data.push_back(function_address);
+            function_address_to_callstack_[function_address].insert(call_stack.GetHash());
+          } else {
+            resolved_callstack_data.push_back(address);
+          }
```

Comment:
This branch should never occur

Reply:
Why not. We can't be sure to have address infos.

Reply:
Well, you are totally right. I removed that branch and directly retrieved the address.